internal/cli/atlas/deployments/options/deployment_opts_pre_run.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package options
+
+import (
+	"context"
+	"errors"
+	"runtime"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/shirou/gopsutil/v3/host"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func (opts *DeploymentOpts) SelectDeployments(ctx context.Context, projectID string) error {
```

Comment:
[nit] I see most of the logic being duplicated (e.g.  GetAtlasDeployments called in 2 different places, opts.findDeployment logic similar to what we have in displayLocalDeployments and displayAtlasDeployments, opts.LocalDeploymentPreRun called in 2 different places).
Can we simplify the flow and do something like this:
```
if opts.IsAtlasDeploymentType() && !opts.IsCliAuthenticated() {
		return ErrNotAuthenticated
}

var atlasDeployments, localDeployments []Deployment
var atlasErr, localErr error

if !opts.IsAtlasDeploymentType() {
        if localErr = opts.LocalDeploymentPreRun(ctx); localErr == nil {
		localDeployments, localErr = ...
	}
}

if !opts.IsLocalDeploymentType() {
      atlasDeployments, atlasErr = opts.GetAtlasDeployments(projectID)
}

if localErr != nil && atlasErr != nil {
      return fmt.Errorf("Failed to retrieve both Atlas and Local deployments:\n%w\n%w", atlasErr, localErr)
}

if localErr != nil {
      _, _ = log.Warningf("Failed to retrieve Local deployments with: %s", localErr.Error()) 
}

if atlasErr != nil {
      _, _ = log.Warningf("Failed to retrieve Atlas deployments with: %s", atlasErr.Error()) 
}

if opts.DeploymentName == "" {
	return opts.Select(append(localDeployments, atlasDeployments...))
}

return opts.findDeployment(localDeployments, atlasDeployments)

```

Reply:
I find the negating logic above slightly more confusing, besides, we should actually fail if customers set --type atlas,  I will try a mix of both

Reply:
I was more concerned about the code duplication and the different behavior for the cases with Atlas fails compared to Local (podman) failures, no need to replicate the example

Reply:
I think that since they're different paths it should be ok, by doing that we make the code more clear and make sure pre checks are ran where needed. I've tried to refactor slightly at least to avoid that start.go still tries to list containers

Reply:
updated!

Reply:
@tibulca I'll merge but feel free to leave more comments as I'll open more PRs for this task